jwt/api_jws.py

```diff
@@ -105,8 +108,8 @@ def get_algorithm_by_name(self, alg_name: str) -> Algorithm:
     def encode(
         self,
         payload: bytes,
-        key: AllowedPrivateKeys | str | bytes,
-        algorithm: str | None = "HS256",
+        key: AllowedPrivateKeys | PyJWK | str | bytes,
+        algorithm: str | None = DEFAULT_ALGORITHM,
```

Comment:
The `DEFAULT_ALGORITHM` is a bit odd. I wonder if we're better off setting this to None?

Reply:
I agree it's odd.  However, `None` means "no signature algorithm" outside the context of a PyJWK key, so it's incredibly dangerous to encourage that:  One misstep with the `key` value and all your JWTs are unsigned.

(I wanted to use a sentinel for DEFAULT_ALGORITHM, a la https://peps.python.org/pep-0661/, rather than a string literal but I couldn't find a way to do that that satisfied me.)

Reply:
Well maybe “dangerous” is an overstatement.

Reply:
maybe worth breaking that behavior. if you want an unsecured JWT you'd have to use `algorithm="none"`

Reply:
Sounds good to me.